src/bin/scribble.ts

```diff
@@ -550,8 +570,65 @@ if ("version" in options) {
     const groupsMap: Map<string, SourceUnit[]> = new Map();
     const ctxtsMap: Map<string, ASTContext> = new Map();
     const filesMap: Map<string, Map<string, string>> = new Map();
-    const originalFiles: Set<string> = new Set();
-    const instrumentationFiles: Set<string> = new Set();
+
+    let metaDataFile: string;
+    let originalFiles: Set<string>;
+    let instrumentationFiles: Set<string>;
+
+    if (options["instrumentation-metadata-file"] !== undefined) {
+        metaDataFile = options["instrumentation-metadata-file"];
+    } else {
+        const projectRoot = detectProjectRoot(targets);
+
+        if (projectRoot !== undefined) {
+            metaDataFile = join(projectRoot, "instrumentation.scribble.json");
+        } else {
+            error(
+                'Unable to detect project root to place instrumentation metadata file. Use "--instrumentation-metadata-file" option to explictly specify location.'
+            );
+        }
+    }
+
+    const isMetaDataFile = fse.existsSync(metaDataFile);
+
+    const instrumentationMarker =
+        "/// This file is auto-generated by Scribble and shouldn't be edited directly.\n" +
+        "/// Use --disarm prior to make any changes.\n";
+
+    /**
+     * In disarm mode we don't need to instrument - just replace the instrumented files with the `.original` files
+     */
+    if (options["disarm"]) {
+        if (isMetaDataFile) {
+            const { original, instrumentation } = getFilesInMetaData(metaDataFile);
+
+            originalFiles = new Set(original);
+            instrumentationFiles = new Set(instrumentation);
+        } else {
+            /**
+             * @todo: Can not throw here due to previous instrumentation will remain in projects.
+             */
+
+            originalFiles = new Set();
+            instrumentationFiles = new Set();
```

Comment:
This case is still unclear, especially if user update Scribble to latest while have some project armed and will not be able to disarm if do not have metadata file somewhere.